File: java/src/main/java/mrk/hackerrank/HackerRank05.java
package mrk.hackerrank;

import java.util.Arrays;
import java.util.stream.LongStream;

//https://www.hackerrank.com/challenges/mini-max-sum/problem
public class HackerRank05 {

    private LongStream createLongStream(int[]arr){
        return Arrays.stream(arr).mapToLong(i -> i);
    }

    public String miniMaxSum(int[] arr) {
        int[] sorted = arr.clone();
        Arrays.sort(sorted);
        return String.format("%d %d", createLongStream(sorted).limit(4).sum(), createLongStream(sorted).skip(1).sum());
    }

}

File: kotlin/src/main/kotlin/HackerRank05.kt
// https://www.hackerrank.com/challenges/mini-max-sum/problem
fun hackerRank05(arr: Array<Int>): String {
    require(arr.size == 5) { "arr must have length 5, was ${arr.size}" }
    var total = arr[0].toLong()
    var min = arr[0].toLong()
    var max = arr[0].toLong()
    for (i in 1 until arr.size) {
        total += arr[i]
        if (arr[i] < min) min = arr[i].toLong()
        if (arr[i] > max) max = arr[i].toLong()
    }
    return String.format("%d %d", total - max, total - min)
}
